### PR TITLE
db create user_ban/kick

### PR DESCRIPTION
**Pull request by Amelia-exe**
updated files so kicks/bans can also be monitored on DB.

### CODE REVIEW

**Comment by Amelia-exe** [comment]
Let me know if this is how it's meant to be? @Dan6erbond

File: database/models/user_ban.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from ..base import Base
+
+
+class UserBan(Base):
+
+    __tablename__ = 'user_bans'
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    guild_id = Column(Integer, ForeignKey("guilds.id"), nullable=False)
+    guild = relationship("Guild")
+    user_id = Column(Integer, nullable=False)
```
Comment (Dan6erbond) [inline database/models/user_ban.py:15]: I recommend adding a `banned_at` column here as well for statistics:
```py
created_at = Column(DateTime, server_default=func.now())
```

File: database/models/user_kick.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from ..base import Base
+
+
+class UserKick(Base):
+
+    __tablename__ = 'user_kicks'
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    guild_id = Column(Integer, ForeignKey("guilds.id"), nullable=False)
+    guild = relationship("Guild")
+    user_id = Column(Integer, nullable=False)
```
Comment (Dan6erbond) [inline database/models/user_kick.py:15]: I recommend adding a `banned_at` column here as well for statistics:
```py
created_at = Column(DateTime, server_default=func.now())
```

File: database/database.py [hunk 3]
```diff
@@ -26,8 +26,10 @@
...
 
     UserMute.__table__.create(bind=engine)
+    UserKick.__table__.create(bind=engine)
+    UserBan.__table__.create(bind=engine)
 
     Message.__table__.create(bind=engine)
 
     Reaction.__table__.create(bind=engine)
-    ReactionRole.__table__.create(bind=engine)
+    ReactionRole.__table__.create(bind=engine)
```
Comment (Dan6erbond) [inline database/database.py:35]: Good catch on the missing table!

**Review by Dan6erbond — APPROVED** [review]
Great changes! It seems as everything would work and the PR can be merged, but I noticed something that you could add for convenience. 😊

As you have multiple models referencing users/members at this point, I feel two classes, `class User` and `class Member` could be added with relationships pointing to their mutes, bans and kicks. This would allow you to do `user.bans`, and if we look into SQLAlchemy's relationships and advanced parameters we might be able to get the same working for members (limited to a user's bans/kicks/mutes within a guild).

File: database/database.py (hunk 3 above)
Comment (Amelia-exe) [reply database/database.py:35]: 😎

File: database/models/user_kick.py (hunk 2 above)
Comment (Amelia-exe) [reply database/models/user_kick.py:15]: made a `banned_at = Column(DateTime, server_default=func.now())` entry in user_kick.

File: database/models/user_ban.py (hunk 1 above)
Comment (Amelia-exe) [reply database/models/user_ban.py:15]: made a `banned_at = Column(DateTime, server_default=func.now())` entry in file.

File: database/models/user_kick.py (hunk 2 above)
Comment (Amelia-exe) [reply database/models/user_kick.py:15]: kicked_at *

**Comment by Amelia-exe** [comment]
Adding idea:
`I noticed something that you could add for convenience. 😊

As you have multiple models referencing users/members at this point, I feel two classes, class User and class Member could be added with relationships pointing to their mutes, bans and kicks. This would allow you to do user.bans, and if we look into SQLAlchemy's relationships and advanced parameters we might be able to get the same working for members (limited to a user's bans/kicks/mutes within a guild).`
into Features (for the future) when you and me have more time to work on this.

**Pull request merged**